src/lib/Components/Inbox/Conversations/ConversationSnippet.tsx

```diff
@@ -169,7 +169,7 @@ export class ConversationSnippet extends React.Component<Props, any> {
                   <MetadataText>
                     {date}
                   </MetadataText>
-                  <UnreadIndicator />
+                  {conversation.is_last_message_to_user && !conversation.last_message_open && <UnreadIndicator />}
```

Comment:
This is the one line to make the unread indicator accurate (based on the Metaphysics data).